tests/transforms/test_broadcast_expand.py

```diff
@@ -260,43 +233,44 @@ def cost(*params):
         jac = t.jacobian(out, params)
         exp_jac = t.jacobian(exp, params)
 
-        if len(obs) > 1 and size == 1:
-            new_exp_jac = [qml.math.flatten(el) for el in exp_jac]
-            new_jac = [qml.math.flatten(el) for el in jac]
-            assert all(qml.math.allclose(_jac, e_jac) for _jac, e_jac in zip(new_jac, new_exp_jac))
-        else:
-            for _jac, e_jac in zip(jac, exp_jac):
-                if e_jac is None:
-                    assert qml.math.allclose(_jac, 0.0)
-                else:
-                    assert qml.math.allclose(_jac, e_jac)
+        for _jac, e_jac in zip(jac, exp_jac):
+            if e_jac is None:
+                assert qml.math.allclose(_jac, 0.0)
+            else:
+                assert qml.math.allclose(_jac, e_jac)
 
     @pytest.mark.torch
     @pytest.mark.filterwarnings("ignore:Output seems independent of input")
-    @pytest.mark.parametrize("params, size", parameters_and_size)
+    @pytest.mark.parametrize("params", parameters)
     @pytest.mark.parametrize("obs, exp_fn", observables_and_exp_fns)
-    def test_torch(self, params, size, obs, exp_fn):
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
```

Comment:
Added diff_method parametrization to check that Torch + broadcasting expansion works and is diffable.